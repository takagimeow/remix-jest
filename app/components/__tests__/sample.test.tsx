import { render, cleanup } from "@testing-library/react";
import React from "react";

import "@testing-library/jest-dom/extend-expect";
import type { Props } from "../Sample";
import { Sample } from "../Sample";

// テスト実行後にDOMをunmount,cleanupします。
afterEach(cleanup);

describe("Test Sample Component", () => {
	afterEach(() => {
		cleanup();
	});

	it("Snapshot Test", async () => {
		const props: Props = {
			text: "Sample"
		};
		const { asFragment } = render(<Sample {...props} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
