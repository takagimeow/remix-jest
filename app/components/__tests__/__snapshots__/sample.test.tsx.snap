// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Sample Component Snapshot Test 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Sample
    </h1>
  </div>
</DocumentFragment>
`;
